library/src/scala/compiletime/int/package.scala

```diff
@@ -0,0 +1,19 @@
+package scala.compiletime
+
+package object int {
+  type +[X <: Int, Y <: Int] <: Int
+  type -[X <: Int, Y <: Int] <: Int
+  type *[X <: Int, Y <: Int] <: Int
+  type /[X <: Int, Y <: Int] <: Int
+  type %[X <: Int, Y <: Int] <: Int
+
```

Comment:
`String` is `AnyRef`, so it'll have to be just `<: Any` (aka no bound)

Reply:
That's a good idea, I like the name `scala.compiletime.ops`. I've made that change in cb3ab697b8443d5d8de687e244ec935d195b43ba.

As for the bounds, I agree. I think that change is best made in the PR that will introduce support for all singleton types, so I'll leave it as is for now, but refer back to this discussion in the next PR.

Reply:
Having moved everything to `scala.compiletime.ops`, it would make sense to also move `scala.compiletime.S` to `scala.compiletime.ops.S`. However, since that would be a breaking change, I think it's best to keep that for a separate PR and have that discussion there.

Reply:
After thinking about it a little more, I think you're right and we need to set the upper bound using match type, otherwise it will not be possible to use the same operation for different types and place the result where there is an expected upper-bound.